### PR TITLE
[NPR] New extractor for NPR.org mediaplayer

### PR DESCRIPTION
**Pull request by Kaspi**

### CODE REVIEW

File: youtube_dl/extractor/npr.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('id')
+        webpage_url = 'http://www.npr.org/player/v2/mediaPlayer.html?id=' + video_id
+        webpage = self._download_webpage(webpage_url, video_id)
+        key = 'MDAzMzQ2MjAyMDEyMzk4MTU1MDg3ZmM3MQ010'
+        xml_url = 'http://api.npr.org/query?id=%s&apiKey=%s' % (video_id, key)
+
+        config = self._download_xml(xml_url,video_id, note='Downloading XML')
+
+        audio = config.findall('./list/story/audio[@type="standard"]')
```
Comment (dstftw) [inline youtube_dl/extractor/npr.py]: This syntax is not supported in python 2.6. You should use `xpath_*` from `utils`.

File: youtube_dl/extractor/npr.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        'id': '445367719',
+        'ext': 'mp4',
+        'title': 'VEGA INTL. Night School'
+    }
+}
+
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('id')
```
Comment (dstftw) [inline youtube_dl/extractor/npr.py]: Use `self._matching_id`.

File: youtube_dl/extractor/npr.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+            for r in regex:
+                t = song.find(r)
+                if t is not None:
+                    format.append({'format': t.get('type', t.tag),
+                               'url' : t.text})
+
+            entries.append({ "title":song_title,
+                             "id":song_id,
+                             "duration": str(int(song_duration) / 60) +":"+ str(int(song_duration) % 60) ,
```
Comment (dstftw) [inline youtube_dl/extractor/npr.py]: duration must be `int`.

File: youtube_dl/extractor/npr.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                             "formats":format})
```
Comment (dstftw) [inline youtube_dl/extractor/npr.py]: Avoid shadowing built-in names.

File: youtube_dl/extractor/npr.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('id')
+        webpage_url = 'http://www.npr.org/player/v2/mediaPlayer.html?id=' + video_id
+        webpage = self._download_webpage(webpage_url, video_id)
+        key = 'MDAzMzQ2MjAyMDEyMzk4MTU1MDg3ZmM3MQ010'
+        xml_url = 'http://api.npr.org/query?id=%s&apiKey=%s' % (video_id, key)
```
Comment (remitamine) [inline youtube_dl/extractor/npr.py]: you can get json output by appending `&format=json` to the api request url.

**Pull request closed without merging**